chainer/training/extensions/micro_average.py

```diff
@@ -0,0 +1,72 @@
+import collections
+
+from chainer import reporter
+from chainer.training import extension
+from chainer.training import util
+
+
+class MicroAverage(extension.Extension):
```

Comment:
It might be better to name it `MicroAverageAccuracy`, since "micro average" is a concept not specific to accuracy.